trino/client.py

```diff
@@ -1187,7 +1187,8 @@ def __iter__(self) -> Iterator[Tuple["SpooledData", "Segment"]]:
         return self
 
     def __next__(self) -> Tuple["SpooledData", "Segment"]:
-        return self, next(self._segments_iterator)
+        segment = next(self._segments_iterator)
+        return SpooledData(self._encoding, [segment]), segment
```

Comment:
Why not also change `SpooledData` to contain a single segment, rather than a list?

Reply:
Because `SpooledData` can contain multiple segments. I think we should not reuse this object as the iterable object but return another class that has all the required fields. Let's get rid of `Tuple["SpooledData", "Segment"]`.

Reply:
Right, each response can contain multiple segment URLs, so we need the transfer object to reflect that, but the abstraction we expose to the client doesn't need to mirror that.